compiler-plugin/src/main/kotlin/arrow/meta/plugins/patternMatching/Desugar.kt

```diff
@@ -0,0 +1,19 @@
+package arrow.meta.plugins.patternMatching
+
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.BindingTrace
+
+val KtCallExpression.desugar: String
+  get() =
+    firstChild.nextSibling
+      .firstChild.nextSibling
+      .text.replace("_", "person.firstName")
+
+val BindingTrace.desugar
+  get() =
+    entriesFollowing(analysisPatternRules).forEach { entry ->
+      // TODO Write replacement entry
+      val parentCaseExpression = entry.key.parent.parent.parent.parent.parent.parent
```

Comment:
this can be done with findParent or similar

Reply:
Is there a build-in method for this? Haven't pushed it yet, but didn't find one so just made a recursive function to search upwards.